misc/inmanta-workon-register.sh

```diff
@@ -204,13 +211,19 @@ function __inmanta_workon_register_deactivate {
         unset -f inmanta >/dev/null 2>&1
         # no need to restore PS1 because virtualenv_deactivate already does that
 
-        if [ -n "${INMANTA_CONFIG_ENVIRONMENT}" ] ; then
+        if [ -n "${_OLD_INMANTA_CONFIG_ENVIRONMENT:-}" ] ; then
+            # Nested environment case: reset INMANTA_CONFIG_ENVIRONMENT to outer env
```

Comment:
I don't think nested environments are a thing here because we always call `deactivate` before activating the other (like `virtualenv` does). Keeping track of the old vars would be too complicated otherwise. It's more for the user's own env vars.